src/tools/wix/ChainPackageInfo.cs

```diff
@@ -611,7 +611,23 @@ private void ResolveMsiPackage(BinderFileManager fileManager, Dictionary<string,
 
                     if (!Common.IsValidModuleOrBundleVersion(this.Version))
                     {
-                        this.core.OnMessage(WixErrors.InvalidProductVersion(this.PackagePayload.SourceLineNumbers, this.Version, sourcePath));
+                        // not a proper .NET version (i.e., five fields); can we get a valid three-part version number?
```

Comment:
Can we try to get a four-part version number before falling back to three?  Burn does use the fourth part.